Scripts/Runtime/Data/VirtualData/VirtualData.cs

```diff
@@ -0,0 +1,296 @@
+        public static readonly int MAX_ELEMENTS_PER_CHUNK = ChunkUtil.MaxElementsPerChunk<TInstance>();
+
+        internal static VirtualData<TKey, TInstance> Create(params IVirtualData[] sources)
+        {
+            VirtualData<TKey, TInstance> virtualData = new VirtualData<TKey, TInstance>();
+
+            foreach (IVirtualData source in sources)
+            {
+                virtualData.AddSource(source);
+                source.AddResultDestination(virtualData);
+            }
+
+            return virtualData;
+        }
+
+        private readonly HashSet<IVirtualData> m_Sources = new HashSet<IVirtualData>();
+        private readonly HashSet<IVirtualData> m_ResultDestinations = new HashSet<IVirtualData>();
+        private readonly AccessController m_AccessController;
+
+        private UnsafeTypedStream<TInstance> m_Pending;
+        private DeferredNativeArray<TInstance> m_Iteration;
+        private UnsafeHashMap<TKey, TInstance> m_Lookup;
+
+        /// <summary>
+        /// The underlying <see cref="IDeferredNativeArray"/> used for scheduling
+        /// <see cref="IAnvilJobForDefer"/> or <see cref="IAnvilJobBatchDefer"/> jobs.
+        /// </summary>
+        public DeferredNativeArray<TInstance> ArrayForScheduling
+        {
+            get => m_Iteration;
+        }
+
+        AccessController IVirtualData.AccessController
+        {
+            get => m_AccessController;
+        }
+
+        private VirtualData()
+        {
+            m_AccessController = new AccessController();
+            m_Pending = new UnsafeTypedStream<TInstance>(Allocator.Persistent,
+                                                         Allocator.TempJob);
+            m_Iteration = new DeferredNativeArray<TInstance>(Allocator.Persistent,
+                                                             Allocator.TempJob);
+
+            m_Lookup = new UnsafeHashMap<TKey, TInstance>(MAX_ELEMENTS_PER_CHUNK, Allocator.Persistent);
+        }
+
+        protected override void DisposeSelf()
+        {
+            m_Pending.Dispose();
+            m_Iteration.Dispose();
+            m_Lookup.Dispose();
+
+            RemoveFromSources();
+            m_ResultDestinations.Clear();
+            m_Sources.Clear();
+
+            m_AccessController.Dispose();
+
+            base.DisposeSelf();
+        }
+
+        //*************************************************************************************************************
+        // SERIALIZATION
+        //*************************************************************************************************************
+
+        //TODO: Add Serialization and Deserialization functions to hook into our serialization framework
+
+        //*************************************************************************************************************
+        // RELATIONSHIPS
+        //*************************************************************************************************************
```

Comment:
mayyyybe. Hopefully we can do it all from the outside, with extension methods or wrapper objects.

Reply:
Is this comment meant to be for the Serialization?

Reply:
Yes

Reply:
I'll update the TODO to reference the hopeful aspect.